### DISCUSSION TITLE
Handling of servers with multiple protocol version support?

### DISCUSSION THREAD

**Discussion by gavinft** [iceiix/stevenarella#376] [General]
How does stevenarella handle servers that support multiple protocol versions (like Hypixel)? I tried joining Hypixel as a test, and the console gave me a few errors about missing models before the game crashed.

**Comment by skillman623** [comment]
Okay so two things. First thing first- for connecting to servers that support multiple versions just connect with a supported version (see below).
The server sometimes succeeds connection to us.hypixel.net using auto protocol detection.
As for selecting versions currently there is no way to do this from the GUI in the client so you have to use the command line/shell.You can manually specify a version by launching stevenarella with .\stevenarella --protocol-version (version in number form).
The following code launches the client and uses 1.12.2
> .\stevenarella.exe --protocol-version 340
.\stevenarella.sh --protocol-version 340

To match up different protocol versions see /scr/protocol/protocol.rs
        "1.16.1" => 736,
        "1.16" => 735,
        "1.15.2" => 578,
        "1.15.1" => 575,
        "1.14.4" => 498,
        "1.14.3" => 490,
        "1.14.2" => 485,
        "1.14.1" => 480,
        "1.14" => 477,
        "19w02a" => 452,
        "18w50a" => 451,
        "1.13.2" => 404,
        "1.12.2" => 340,
        "1.11.2" => 316,
        "1.11" => 315,
        "1.10.2" => 210,
        "1.9.2" => 109,
        "1.9" => 107,
        "15w39c" => 74,
        "1.8.9" => 47,
        "1.7.10" => 5,
As for the crash, please refer to #377

**Comment by gavinft** [comment]
Thanks. It would be cool to add a manual protocol version for each server to the GUI, so when you add a server to the list, you can select the protocol verssion you would like.

**Reply by PureTryOut** [reply]
I wonder, why as an end user would you care what version is used to join the server? I'd think you would only care that you _can_ join, not how.

**Reply by Vresod** [reply]
most servers that support multiple versions (usually using viaversion) make sure the client supports the features given by replacing old stuff with new stuff and vice versa (notably replacing sword blocking with shields on 1.9+ clients)

**Reply by mpfaff** [reply]
> I wonder, why as an end user would you care what version is used to join the server? I'd think you would only care that you _can_ join, not how.

The experience on hypixel, for example, is very different on 1.8 than on later versions. Major differences include that most (not all) velocity packets are sent as silent explosions instead. Also, the knockback in combat is wildly unpredictable. Imo, it should be up to the user which version to use on a server like hypixel since for games like skyblock, 1.16+ is better, while in games like skywars and bedwars 1.8 is better.
